src/connector.js

```diff
@@ -589,10 +778,32 @@ function filterProtocol( uri ) {
 	return isAbsolute || isRelative ? uri : '';
 }
 
+// Get date converted from GMT (Coveo) to current timezone
+function getDateInCurrentTimeZone( date ){
+	const offset = date.getTimezoneOffset();
+	return new Date( date.getTime() + ( offset * 60 * 1000 ) );
+}
+
+// get a short date format like YYYY-MM-DD
+function getShortDateFormat( date ){
+	let currentTZDate = getDateInCurrentTimeZone( date );
+	return currentTZDate.toISOString().split( 'T' )[ 0 ];
+}
+
+// get a long date format like May 21, 2024
+function getLongDateFormat( date, lang ){
+	let currentTZDate = getDateInCurrentTimeZone( date );
+	if ( lang === 'en' ) {
+		return monthsEn[ currentTZDate.getMonth() ] + " " + currentTZDate.getDate() + ", " + currentTZDate.getFullYear();
+	}
+
+	return currentTZDate.getDate() + " " + monthsFr[ currentTZDate.getMonth() ] + " " + currentTZDate.getFullYear();
+}
```

Comment:
Check to leverage wb utilities

https://github.com/wet-boew/wet-boew/blob/master/src/core/helpers.js#L1209-L1299

And JS core like: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString